servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/NoopTransportObserver.java

```diff
@@ -43,260 +45,111 @@ private NoopTransportObserver() {
 
     @Override
     public ConnectionObserver onNewConnection(@Nullable final Object localAddress, final Object remoteAddress) {
-        return NoopConnectionObserver.INSTANCE;
+        return onNewConnection();
     }
 
     /**
      * Noop version of {@link ConnectionObserver}.
      */
     public static final class NoopConnectionObserver implements ConnectionObserver {
 
-        public static final ConnectionObserver INSTANCE = new NoopConnectionObserver();
+        // Here and in all other INSTANCE cases below we call original methods to get the reference of the pkg-private
+        // NOOP variants from transport-api module. That way we can make sure that checking for NOOP reference can be
+        // used as a way to skip observability wiring in cases where it adds overhead.
+        public static final ConnectionObserver INSTANCE = NoopTransportObserver.INSTANCE.onNewConnection();
```

Comment:
This is kind of hairy to understand. I suppose that the fallout is that we unnecessarily feed events to a noop observer so it's not so bad.

Reply:
I guess I should have noted what I wanted from this: it's just a comment for now. Maybe we can find a better way in a follow up. I think you suggested we could allow returning null, but that deserves more thought and a separate PR.